packages/edit-site/src/components/global-styles/icon-with-current-color.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/components';
+
+export function IconWithCurrentColor( { className, ...props } ) {
```

Comment:
As much as I would like to see this kind of thing supported by default in `Icon` itself, that component is extremely polymorphic right now so it might not actually be a good idea. Maybe local wrappers components are the way to go 🤷 @ciampo

Reply:
Agreed. Moved to #40102